app/controllers/api/base_controller.rb

```diff
@@ -94,4 +99,8 @@ def authorize_if_got_token!(*scopes)
   def set_cache_headers
     response.headers['Cache-Control'] = 'no-cache, no-store, max-age=0, must-revalidate'
   end
+
+  def disallow_unauthenticated_api_access?
+    whitelist_mode?
```

Comment:
I would probably tie that to the `authorized_fetch` mode more than the whitelist mode